jwst/source_catalog/source_catalog.py

```diff
@@ -144,24 +144,28 @@ def make_source_catalog(model, kernel_fwhm, kernel_xsize, kernel_ysize,
     catalog['orientation_sky'] = ((270. - rot +
                                    catalog['orientation'].value) * u.deg)
 
-    # define AB mag and AB mag error
+    # define flux in microJanskys
+    nsources = len(catalog)
     pixelarea = model.meta.photometry.pixelarea_arcsecsq
     if pixelarea is None:
-        micro_Jy = 0.0
+        micro_Jy = np.full(nsources, np.nan)
```

Comment:
If you don't specify the data type, will this give you the data type that you expect?